### PR TITLE
CPDP-2800: Delta charges

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,3 @@
 # cgt-property-disposals
 
-For more context of Capital Gain Tax on UK Property (CGT), see [here](https://github.com/hmrc/cgt-property-disposals-frontend#cgt-property-disposals-frontend).
```
Comment (AndyHWChung) [inline README.md:3]: the README text should be restored

File: app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala [hunk 2]
```diff
@@ -113,6 +123,79 @@ class DefaultReturnsService @Inject() (
     } yield returnResponse
   }
 
+  private def deltaCharge(
+    response: DesSubmitReturnResponse,
+    cgtReference: CgtReference,
+    fromDate: LocalDate,
+    toDate: LocalDate
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, Option[DeltaCharge]] =
+    response.ppdReturnResponseDetails.chargeReference match {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala:132]: if the return is not an amend return we can return `None` straight away - delta charges only apply to amends

File: app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala [hunk 3]
```diff
@@ -113,6 +123,79 @@ class DefaultReturnsService @Inject() (
...
+      case Some(chargeReference) =>
+        lazy val charge = for {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala:134]: you don't need the `laazy val` here, you can just have the for-comprehension by itself

File: app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala [hunk 4]
```diff
@@ -113,6 +123,79 @@ class DefaultReturnsService @Inject() (
...
+          desFinancialData <- getDesFinalcialData(cgtReference, fromDate, toDate)
+          charge           <- EitherT.fromEither[Future](findDeltaCharge(desFinancialData.financialTransactions, chargeReference))
+        } yield charge
+
+        charge
+      case _                     => EitherT.pure(None)
+    }
+
+  private def getDesFinalcialData(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala:143]: this is the same code that appears elsewhere - it'd be better to make the original code not private and reuse it here

Comment (rambabu-posa) [reply app/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsService.scala:143]: yes right, I had same plan to write a function and reuse at both places but forgot that one.

**Comment by AndyHWChung** [comment]
is there a spec which tests that delta charge is returned properly? I can't see one but I could just be missing it

File: test/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsServiceSpec.scala [hunk 5]
```diff
@@ -1460,6 +1460,281 @@ class ReturnsServiceSpec extends WordSpec with Matchers with MockFactory {
...
+
+        }
+
+        "there are two DesFinancialTransactions which matches return's charge reference" in {
+
+          val formBundleId    = "804123737752"
+          val chargeReference = "XCRG9448959757"
+
+          val submitReturnRequest    = sample[SubmitReturnRequest].copy(
+            originalReturnFormBundleId = None
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsServiceSpec.scala:1594]: shouldn't this be `Some(..)`? delta charges should only apply when it's an amend return

Comment (AndyHWChung) [reply test/uk/gov/hmrc/cgtpropertydisposals/service/returns/ReturnsServiceSpec.scala:1594]: how does this test pass if it's not an amend return and you're expecting a `DeltaCharge` at the end of it?

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**